### ISSUE TITLE
[analyzer] Missing `FieldDeclaration.covariantKeyword` in ToSourceVisitor

### ISSUE THREAD

**Issue by incendial** [dart-lang/sdk#55273]
Probably should be before or after the static keyword, [here](https://github.com/dart-lang/sdk/blob/main/pkg/analyzer/lib/src/dart/ast/to_source_visitor.dart#L480).

Additionally, the `question` token is not added [here](https://github.com/dart-lang/sdk/blob/main/pkg/analyzer/lib/src/dart/ast/ast.dart#L7152) and in several other places (can open a separate issue, if needed).

**Comment by keertip** [comment]
@incendial, did you run into issues while trying to use the analyzer API?

**Comment by bwilkerson** [comment]
Yes. This impacts the `toSource()` method on `AstNode`. That method only promises to approximate valid source, so clients shouldn't be depending on it, which means that this shouldn't be a high priority issue.

**Comment by incendial** [comment]
It also affects the `toString` call on any AST node (which has no disclaimer for being "approximate valid source") and is sometimes used in rules (both as `toSource` and `toString`, in, for example, interpolations).

> did you run into issues while trying to use the analyzer API?
I don't use the analyzer as a package (and already have a fix for it), so depending on your definition of an API the answer may vary. But this is a bug that affect the public API.
